src/Microsoft.Health.Fhir.Api/Features/Filters/ValidateContentTypeFilterAttribute.cs

```diff
@@ -44,21 +44,28 @@ public override async Task OnActionExecutionAsync(ActionExecutingContext context
             {
                 if (contextHttpContext.Request.Headers.TryGetValue(HeaderNames.ContentType, out StringValues headerValue))
                 {
-                    var resourceFormat = ContentType.GetResourceFormatFromContentType(headerValue[0]);
-
-                    if (!await _contentTypeService.IsFormatSupportedAsync(resourceFormat))
+                    try
                     {
-                        string routeName = context.ActionDescriptor?.AttributeRouteInfo?.Name;
+                        var resourceFormat = ContentType.GetResourceFormatFromContentType(headerValue[0]);
 
-                        switch (routeName)
+                        if (!await _contentTypeService.IsFormatSupportedAsync(resourceFormat))
                         {
-                            case RouteNames.SearchResourcesPost:
-                            case RouteNames.SearchAllResourcesPost:
-                                break;
-                            default:
-                                throw new UnsupportedMediaTypeException(Resources.UnsupportedContentTypeHeader);
+                            string routeName = context.ActionDescriptor?.AttributeRouteInfo?.Name;
+
+                            switch (routeName)
+                            {
+                                case RouteNames.SearchResourcesPost:
+                                case RouteNames.SearchAllResourcesPost:
+                                    break;
+                                default:
+                                    throw new UnsupportedMediaTypeException(Resources.UnsupportedContentTypeHeader);
+                            }
                         }
                     }
+                    catch (NullReferenceException)
```

Comment:
We generally don't want to be catching this kind of exception, but instead should use a conditional expression before dereferencing.